core/state_processor.go

```diff
@@ -232,6 +232,11 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	// Finalize the block, applying any consensus engine specific extras (e.g. block rewards)
 	p.engine.Finalize(p.bc, header, statedb, block.Transactions(), block.Uncles())
 
+	if block.NumberU64()%100 == 0 {
+		stateRoot := statedb.GetTrie().Hash()
```

Comment:
`statedb.GetTrie()` will return the corresponding trie for the current height.
After the transition started, this will be a `TransitionTrie` which `(*TransitionTrie).Hash()` returns `t.overlay.Hash()`.